pkg/controller/handler.go

```diff
@@ -245,7 +245,7 @@ func (p *Propeller) Handle(ctx context.Context, namespace, name string) error {
 		t := p.metrics.RoundTime.Start(ctx)
 		mutatedWf, err := p.TryMutateWorkflow(ctx, w)
 
-		if wfClosureCrdFields != nil {
+		if wfClosureCrdFields != nil && mutatedWf != nil {
```

Comment:
Can we instead just move this down where the `err != nil` and `mutatuedWf == nil` check are already occurring?

```suggestion
		if err != nil {
			// NOTE We are overriding the deepcopy here, as we are essentially ingnoring all mutations
			// We only want to increase failed attempts and discard any other partial changes to the CRD.
			mutatedWf = RecordSystemError(w, err)
			p.metrics.SystemError.Inc(ctx)
		} else if mutatedWf == nil {
			logger.Errorf(ctx, "Should not happen! Mutation resulted in a nil workflow!")
			return nil
		} else {
			if wfClosureCrdFields != nil {
				// strip data populated from WorkflowClosureReference
				w.SubWorkflows, w.Tasks, w.WorkflowSpec = nil, nil, nil
				mutatedWf.SubWorkflows, mutatedWf.Tasks, mutatedWf.WorkflowSpec = nil, nil, nil
			}

			if !w.GetExecutionStatus().IsTerminated() {
				// No updates in the status we detected, we will skip writing to KubeAPI
				if mutatedWf.Status.Equals(&w.Status) {
					logger.Info(ctx, "WF hasn't been updated in this round.")
					t.Stop()
					return nil
				}
			}
			if mutatedWf.GetExecutionStatus().IsTerminated() {
				// If the end result is a terminated workflow, we remove the labels
				// We add a completed label so that we can avoid polling for this workflow
				SetCompletedLabel(mutatedWf, time.Now())
				ResetFinalizers(mutatedWf)
			}
		}
```